pkg/gce-pd-csi-driver/utils.go

```diff
@@ -212,3 +217,66 @@ func containsZone(zones []string, zone string) bool {
 
 	return false
 }
+
+// CodeForError returns a pointer to the grpc error code that maps to the http
+// error code for the passed in user googleapi error or context error. Returns
+// codes.Internal if the given error is not a googleapi error caused by the user.
+// The following http error codes are considered user errors:
+// (1) http 400 Bad Request, returns grpc InvalidArgument,
+// (2) http 403 Forbidden, returns grpc PermissionDenied,
+// (3) http 404 Not Found, returns grpc NotFound
+// (4) http 429 Too Many Requests, returns grpc ResourceExhausted
+// The following errors are considered context errors:
+// (1) "context deadline exceeded", returns grpc DeadlineExceeded,
+// (2) "context canceled", returns grpc Canceled
+func CodeForError(err error) *codes.Code {
```

Comment:
Looks like this bit of code was moved to `common/utils.go` in #1227 and I believe it's packported together with Matt's change all the way to 1.9. Have we checked if adding it here again is the right thing to do?

Reply:
Thanks for pointing this out @leiyiz. After some digging I found #1240 actually effectively backports #1230, but not via the same commit. So we don't need this CL after all.